taliforth.asm

```diff
@@ -560,6 +561,28 @@ error:
 
                 jsr print_common
                 jsr xt_cr
+                pla                     ; recover error number
+
+                cmp #err_underflow      ; should we display return stack?
+                bne _no_underflow
+
+                lda #err_returnstack
```

Comment:
Having the strings numbered makes it super simple to do translations, so all of the strings should have a number.

I don't see you using err_returnstack here, but I don't see it defined anywhere in this pull request - it should be defined in strings.asm as 13.  If this is assembling without error for you, you likely have it already and just missed it when pushing your latest changes.  

I'm fine with this method of handling errors with a special check for underflow, so I'll probably merge this once err_returnstack appears.

Reply:
sorry, not sure what happened there.  i refactored print_error into the print section and used twice for the return stack case.  Sample output:

```
Tali Forth 2 comes with absolutely NO WARRANTY
Type 'bye' to exit
. Data stack underflow
Return stack: 38 8C 42 8F 13 D8 B0 80
3 0 / Division by zero
: foo drop ;  ok
: bar foo ;  ok
' foo . 2059  ok
' bar . 2076  ok
bar Data stack underflow
Return stack: 0D 08 1E 08 42 8F 13 D8 B0 80
$80d u. 2061  ok
$81e u. 2078  ok
```